include/odp/api/spec/pool.h

```diff
@@ -228,6 +243,12 @@ typedef struct odp_pool_param_t {
 			    capability pkt.max_headroom.
 			    Use zero if headroom is not needed. */
 			uint32_t headroom;
+
+			/* Random Early Detection configuration */
+			odp_red_param_t red;
+
+			/* Back Pressure configuration */
+			odp_bp_param_t bp;
 		} pkt;
```

Comment:
Need to clarify under what conditions RED (and BP) apply to pools. If an application calls `odp_packet_alloc()` that should either succeed or fail. "Drop probability" doesn't make any obvious sense here. Tying these concepts to `odp_pktio_t` configurations seems more in keeping with what's needed, but I understand the difficulties that that caused in the previous PR.

Perhaps tying this to a `odp_cos_t` might work? An `odp_cos_t` has an associated queue and pool and these can then be viewed as admission controls to packets being added to a CoS. That would mean that these features would only be applicable if the classifier is being used, but is that a significant drawback since we want to encourage its use anyway?

Reply:
The previous PR was handling this RED and BP at cos level. There were some concerns from NXP since CoS is a logical entity and application configuration should make sure that multiple CoS with different RED configuration should not direct to the same pool or queue.

HW implements the RED and BP effectively either at the pool or the queue.
Since a pool represents the real bottleneck for resource exhausion and packet drop.
 We can discuss further in todays Public call.